l10n_ch_pain_base/models/account_payment_order.py

```diff
@@ -68,7 +68,8 @@ def generate_party_agent(
             if bank_line.local_instrument == 'CH01':
                 # Don't set the creditor agent on BVR/CH01 payments
                 return True
-            elif not partner_bank.bank_bic:
+            elif not partner_bank.bank_bic and \
+                    self.payment_method_id.bic_required:
```

Comment:
As we are in a `@api.model` the reference to `self.payment_method_id` is not supposed to work here.

Reply:
Answer: It is working as the caller `generate_party_block()` is decorated with `@api.multi` so `self` represents a recordset, but it's not expected as input.
I'll fix this in `bank-payment` modules while porting this code to 12.0, I don't think it worth to pass time to "fix it" on 9.0.